### ISSUE TITLE
[feature] Security: protect accounts with 2FA (or MFA)

### ISSUE THREAD

**Issue by igalic** [superseriousbusiness/gotosocial#395]
currently, authentication is entirely password based.
In an ideal world, GtS would support 2FA (or MFA), that is: two factor authentication, or multi-factor authentication.

Personally, I use\* Yubikeys, which implement a number of standards: https://www.yubico.com/authentication-standards/

Yubikeys also support TOTP, thru [Yubico Authenticator](https://www.yubico.com/products/yubico-authenticator/)
(which i trust slightly more than solely relying on Google Authenticator on a Google phone)

____________
\*I also wear them as fashion accessories.

**Comment by tsmethurst** [comment]
I'm gonna put this under [Future Tech](https://github.com/superseriousbusiness/gotosocial/labels/future%20tech) because it would be cool to have, but it's probably not something we'll implement straight away.

**Comment by cortices** [comment]
May be useful to check out the PR on Gitea's codebase that implemented the webauthn framework, the modern browser U2F/FIDO stack, for how it's implemented.
https://github.com/go-gitea/gitea/pull/17957

**Comment by robin24** [comment]
Hi,

Just curious, have there been any news / updates around this since the issue was opened back in 2022?
I'd love to deploy this for an organization that I work with, as it seems perfect for a small instance with only a handful user accounts.
MFA is a strict requirement for them though which I'm 100% in agreement with, so would be great to know if GTS has added support for this yet in some capacity.

Thanks!

**Comment by tsmethurst** [comment]
Yes! It was actually on our internal / grant planning for the end of this month, but it looks likely it'll be something for next month instead as we're a bit behind unfortunately (as always :'))

**Comment by robin24** [comment]
Thanks so much for the heads-up!

Glad to hear this is on the roadmap and will be worked on in the near future, I'll be keeping an eye on this issue for sure.

Thanks as well for pointing out OIDC as a workaround. I actually have already discussed this with my team, but consensus was that it would add too much complexity given the small number of users so probably won't go down that path. Really appreciate the idea however, maybe useful to someone else coming across this issue :-).

**Comment by skobkin** [comment]
I've opened a followup issue for WebAuthn: #4046
